### DISCUSSION TITLE
Discord authentication strategy - "invalid oauth2 redirect URI" after domain change

### DISCUSSION THREAD

**Discussion by Maxzor** [requarks/wiki#3618] [Error / Bug Report]
I have first setup the authentication strategy using the VPS bare-foot http URI.
Discord login for new users was functioning fine.

I then have setup a "real" domain at https://theholywitcher3moddingbible.org
I updated in Wiki.js, from the admin panel the domain, and in Discord OAuth2 the redirect URI as newly displayed by wiki.js

You can check by yourself on the above domain, when trying to log in with discord, there is a 400 GET.
The domain seems duplicated in the request, among maybe other issues.

Emitted by wikijs
```
https://discord.com/oauth2/authorize?prompt=none&response_type=code&redirect_uri=https%3A%2F%2Ftheholywitcher3moddingbible.org%2Flogin%2Ftheholywitcher3moddingbible.org%2Flogin%2Ff5f0cc73-6793-421a-be60-709922781bbe%2Fcallback&scope=identify%20email%20guilds&client_id=760646313750298625
```
Discord scope example
```
https://discord.com/api/oauth2/authorize?client_id=760646313750298625&redirect_uri=http%3A%2F%2Ftheholywitcher3moddingbible.org%2Flogin%2Ff5f0cc73-6793-421a-be60-709922781bbe%2Fcallback&response_type=code&scope=identify%20email
```
![image](https://user-images.githubusercontent.com/16599683/95006378-44de7e80-0604-11eb-8204-1229a72aeb06.png)

Wiki.js 2.5.144

**Comment by Maxzor** [comment]
This is the culprit.
/wiki/node_modules/passport-oauth2/lib/strategy.js :
```js
      // The callback URL is relative, resolve a fully qualified URL from the
      // URL of the originating request.
      callbackURL = url.resolve(utils.originalURL(req, { proxy: this._trustProxy }), callbackURL);
```

**Comment by Maxzor** [comment]
passing a relative redirect url, e.g. with server/core/auth.js l84 :  `stg.config.callbackURL = '${stg.key}/callback'` , the oauth2 exchange seems to be solved here.
but it is passport-oauth2 now that complains that the redirect uri is wrong

**Comment by Maxzor** [comment]
I came to the conclusion that the library is bugged. Did not report upstream yet.
Just :
- comment out this line https://github.com/Requarks/wiki/issues/2536#issuecomment-704630856
- make this fqdn and prefix the protocol in there https://github.com/Requarks/wiki/issues/2536#issuecomment-704674036

and for discord at least this is fine.

**Comment by ozdeadmeat** [comment]
Has the Discord Authenticator issues been resolved?
